app/controllers/patients_controller.rb

```diff
@@ -86,6 +102,27 @@ def set_programme_statuses
           needs_consent_request_failed
           needs_consent_follow_up_requested
         ]
+    end
+  end
+
+  def set_show_patient_programme_status
```

Comment:
Adding these `before_action` methods works, but I think it might be simpler to put these directly in the view actually, we only usually set a before action if the logic is particularly complex.